scripts/validation/validate-versions-staged-files.js

```diff
@@ -0,0 +1,62 @@
+const { join, dirname } = require("path");
+const { promisify } = require("util");
+const { exec, execFile } = require("child_process");
+
+const execAsync = promisify(exec);
+const execFileAsync = promisify(execFile);
+
+preCommit().catch(error => {
+    console.error(error);
+    process.exit(1);
+});
+
+async function preCommit() {
+    const [{ stdout: lernaPackages }, { stdout: stagedFiles }] = await Promise.all([
+        execAsync("./node_modules/.bin/lerna ls --json --all"),
+        execAsync("git diff --staged --name-only")
+    ]);
+    const packages = JSON.parse(lernaPackages.trim());
+    const staged = stagedFiles.trim().split("\n");
+    const changedWidgetPackages = packages
+        .filter(({ location }) =>
+            staged.some(
+                changedFilePath =>
+                    dirname(join(process.cwd(), changedFilePath)).includes(location) &&
+                    changedFilePath
+                        .split("/")
+                        .pop()
+                        .match(/package\.(json|xml)$/)
+            )
+        )
+        .filter(({ location }) => location.match(/(pluggable|custom)Widgets/));
+
+    if (changedWidgetPackages.length) {
+        const validationPromises = [];
+
+        for (const changedWidgetPackage of changedWidgetPackages) {
+            validationPromises.push(
+                new Promise((resolve, reject) => {
+                    execFileAsync("node", [join(process.cwd(), "scripts/validation/validate-version.js")], {
+                        cwd: changedWidgetPackage.location
+                    })
+                        .then(() => resolve())
+                        .catch(error => reject(error));
+                })
+            );
+        }
+
+        const validationResults = await Promise.all(
```

Comment:
What do you think about catching errors on the promise all? Is there a specific reason to catch everything inside already?

Reply:
I thought it best to validate all staged widget files. With `Promise.all` only the first rejection is returned. This is why I map, catch, then return a string (for rejections) and implictly return the undefined fulfilled promises, to essentially get an array of resolved promises where errors are denoted by a string.

Reply:
I think [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) is what you need 🙂

Reply:
Very naaaaize!